import-dbass-dashboard.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+
+# Grafana dashboard importer script.
+
+import base64
+import binascii
+import hashlib
+import json
+import os
+import random
+import sqlite3
+import string
+import subprocess
+import sys
+import time
+import httplib
+import requests
+
+GRAFANA_DB_DIR             = sys.argv[1] if len(sys.argv) > 1 else '/srv/grafana'
+HOST                       = 'http://127.0.0.1:3000'
+DBAAS                      = os.getenv('PERCONA_TEST_DBAAS') is not None
+
+
+def grafana_headers(api_key):
+    """
+    Returns HTTP headers for all requests to Grafana.
+    """
+
+    return {'Authorization': 'Bearer %s' % (api_key,), 'Content-Type': 'application/json'}
+
+
+def get_api_key():
+    """
+    Generates a new API key and returns its name, representation for API, and representation for DB.
+
+    Keep in sync with Grafana implementation:
+    * https://sourcegraph.com/github.com/grafana/grafana/-/blob/pkg/api/apikey.go
+    * https://sourcegraph.com/github.com/grafana/grafana/-/blob/pkg/components/apikeygen/apikeygen.go
+    * https://sourcegraph.com/github.com/grafana/grafana/-/blob/pkg/util/encoding.go
+    """
+
+    alphanum = string.digits + string.ascii_uppercase + string.ascii_uppercase.lower()
+    name = 'PMM Import ' + ''.join(random.choice(alphanum) for _ in range(16))
+    key = ''.join(random.choice(alphanum) for _ in range(32))
+    api_key = base64.b64encode(json.dumps({'k': key, 'n': name, 'id': 1}))
+    db_key = binascii.hexlify(hashlib.pbkdf2_hmac('sha256', key, name, 10000, 50))
+    return (name, api_key, db_key)
+
+
+def start_grafana():
+    res = None
+    if os.path.exists('/usr/bin/supervisorctl'):
+        res = subprocess.call(["/usr/bin/supervisorctl", "start", "grafana"])
+    else:
+        res = subprocess.call(["/bin/systemctl", "start", "grafana-server"])
+    print ' * Grafana start: %r.' % (res,)
+
+
+def stop_grafana():
+    res = None
+    if os.path.exists('/usr/bin/supervisorctl'):
+        res = subprocess.call(["/usr/bin/supervisorctl", "stop", "grafana"])
+    else:
+        res = subprocess.call(["/bin/systemctl", "stop", "grafana-server"])
+    print ' * Grafana stop: %r.' % (res,)
+
+    # wait for full stop
+    time.sleep(5)
+
+
+def wait_for_grafana_start():
+    sys.stdout.write(' * Waiting for Grafana to start')
+    sys.stdout.flush()
+    for _ in xrange(60):
+        try:
+            requests.get('%s/api/datasources' % HOST, timeout=0.1)
+        except (requests.exceptions.ConnectionError, requests.exceptions.ReadTimeout):
+            sys.stdout.write('.')
+            sys.stdout.flush()
+            time.sleep(1)
+        else:
+            print
+            return
+    print "\n * Grafana is unable to start correctly"
+    sys.exit(-1)
+
+
+def add_api_key(name, db_key):
```

Comment:
Can we reuse this function from a sibling file?

Reply:
I have get rid of an additional script and added changes into import-dashboards.py